src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml.cs

```diff
@@ -89,6 +106,114 @@ public RecentFilesWidget()
+					new List<ListedItem>() { new ListedItem(null!) { ItemPath = item.RecentPath } }, shiftPressed: shiftPressed, showOpenMenu: false, default);
+				if (!UserSettingsService.AppearanceSettingsService.MoveShellExtensionsToSubMenu)
+				{
+					var (_, secondaryElements) = ItemModelListToContextFlyoutHelper.GetAppBarItemsFromModel(shellMenuItems);
+					if (!secondaryElements.Any())
+						return;
+
+					var openedPopups = Microsoft.UI.Xaml.Media.VisualTreeHelper.GetOpenPopups(App.Window);
+					var secondaryMenu = openedPopups.FirstOrDefault(popup => popup.Name == "OverflowPopup");
+
+					var itemsControl = secondaryMenu?.Child.FindDescendant<ItemsControl>();
+					if (itemsControl is not null)
+					{
+						var maxWidth = itemsControl.ActualWidth - Constants.UI.ContextMenuLabelMargin;
+						secondaryElements.OfType<FrameworkElement>()
+										 .ForEach(x => x.MaxWidth = maxWidth); // Set items max width to current menu width (#5555)
+					}
+
+					itemContextMenuFlyout.SecondaryCommands.Add(new AppBarSeparator());
+					secondaryElements.ForEach(i => itemContextMenuFlyout.SecondaryCommands.Add(i));
+				}
+				else
+				{
+					var overflowItems = ItemModelListToContextFlyoutHelper.GetMenuFlyoutItemsFromModel(shellMenuItems);
+					if (itemContextMenuFlyout.SecondaryCommands.FirstOrDefault(x => x is AppBarButton appBarButton && (appBarButton.Tag as string) == "ItemOverflow") is not AppBarButton overflowItem)
+						return;
+
+					var flyoutItems = (overflowItem.Flyout as MenuFlyout)?.Items;
+					if (flyoutItems is not null)
+						overflowItems.ForEach(i => flyoutItems.Add(i));
+					overflowItem.Visibility = overflowItems.Any() ? Visibility.Visible : Visibility.Collapsed;
+				}
+			}
+			catch { }
+		}
+
+		private List<ContextMenuFlyoutItemViewModel> GetRecentItemMenuItems(RecentItem item)
+		{
+			return new List<ContextMenuFlyoutItemViewModel>()
+			{
+				new ContextMenuFlyoutItemViewModel()
+				{
+					Text = "RecentItemRemove/Text".GetLocalizedResource(),
+					Glyph = "\uF117",
+					GlyphFontFamilyName = "CustomGlyph",
+					Command = RemoveRecentItemCommand,
+					CommandParameter = item,
+					ShowItem = true
+				},
+				new ContextMenuFlyoutItemViewModel()
+				{
+					Text = "RecentItemClearAll/Text".GetLocalizedResource(),
+					Glyph = "\uF113",
+					GlyphFontFamilyName = "CustomGlyph",
+					Command = ClearAllItemsCommand,
+					ShowItem = true
+				},
+				new ContextMenuFlyoutItemViewModel()
+				{
+					Text = "RecentItemOpenFileLocation/Text".GetLocalizedResource(),
+					Glyph = "\uE737",
+					Command = OpenFileLocationCommand,
+					CommandParameter = item,
+					ShowItem = true
+				},
+				new ContextMenuFlyoutItemViewModel()
+				{
+					Text = "ShowMoreOptions".GetLocalizedResource(),
+					Glyph = "\xE712",
+					Items = new List<ContextMenuFlyoutItemViewModel>(),
+					ID = "ItemOverflow",
+					Tag = "ItemOverflow",
+					IsHidden = true,
+				}
+			}.Where(x => x.ShowItem).ToList();
```

Comment:
You can remove `Where` since here `ShowItem` is a constant. 
Actually, you can remove `ShowItem = true` since that is its default value.